cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceContext.java

```diff
@@ -130,6 +143,25 @@
   void setInput(String datasetName, Dataset dataset);
 
   /**
+   * Overrides the input configuration of this MapReduce job to read from the specified input.
+   * @param input the input to be used
+   */
+  void setInput(Input input);
```

Comment:
So this becomes asymmetric to the output, which we only have `addOutput`? I think we had the discussion about whether we should have both `set` and `add` or just `add` when we were dealing with multiple outputs?